# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/ShadowLayers.cpp
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 gfx/layers/ipc/ShadowLayers.cpp
--- a/gfx/layers/ipc/ShadowLayers.cpp	Tue Apr 07 15:17:14 2020 +0000
+++ b/gfx/layers/ipc/ShadowLayers.cpp	Tue Apr 07 15:17:47 2020 +0000
@@ -19,7 +19,6 @@
 #include "ipc/IPCMessageUtils.h"  // for gfxContentType, null_t
 #include "IPDLActor.h"
 #include "mozilla/Assertions.h"  // for MOZ_ASSERT, etc
-#include "mozilla/dom/TabGroup.h"
 #include "mozilla/gfx/Point.h"                  // for IntSize
 #include "mozilla/layers/CompositableClient.h"  // for CompositableClient, etc
 #include "mozilla/layers/CompositorBridgeChild.h"
@@ -195,9 +194,8 @@
       mIsFirstPaint(false),
       mNextLayerHandle(1) {
   mTxn = new Transaction();
-  if (TabGroup* tabGroup = mClientLayerManager->GetTabGroup()) {
-    mEventTarget = tabGroup->EventTargetFor(TaskCategory::Other);
-  }
+  mEventTarget = GetMainThreadSerialEventTarget();
+
   MOZ_ASSERT(mEventTarget || !XRE_IsContentProcess());
   mActiveResourceTracker = MakeUnique<ActiveResourceTracker>(
       1000, "CompositableForwarder", mEventTarget);